File: back/src/main/java/com/back/domain/trend/snapshot/entity/DailyCooccurrenceCount.java
package com.back.domain.trend.snapshot.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "daily_cooccurrence_count",
        uniqueConstraints = @UniqueConstraint(columnNames ={"date","keyword_a","keyword_b"}))
public class DailyCooccurrenceCount {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDate date;

    @Column(name = "keyword_a")
    private String keywordA;

    @Column(name = "keyword_b")
    private String keywordB;

    private long frequency;

    public void updateFrequency(long frequency) {
        this.frequency = frequency;
    }

    public DailyCooccurrenceCount(LocalDate date, String keywordA, String keywordB, long frequency) {
        this.date = date;
        this.keywordA = keywordA;
        this.keywordB = keywordB;
        this.frequency = frequency;
    }
}

File: back/src/main/kotlin/com/back/domain/trend/snapshot/entity/DailyCooccurrenceCount.kt
package com.back.domain.trend.snapshot.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import java.time.LocalDate

@Entity
@Table(
    name = "daily_cooccurrence_count",
    uniqueConstraints = [UniqueConstraint(columnNames = ["date", "keyword_a", "keyword_b"])]
)
class DailyCooccurrenceCount(
    date: LocalDate,
    keywordA: String,
    keywordB: String,
    frequency: Long
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        protected set

    var date: LocalDate = date
        protected set

    @Column(name = "keyword_a")
    var keywordA: String = keywordA
        protected set

    @Column(name = "keyword_b")
    var keywordB: String = keywordB
        protected set

    var frequency: Long = frequency
        protected set

    fun updateFrequency(frequency: Long) {
        this.frequency = frequency
    }
}
